src/components/common/dashboard/Dashboard.vue

```diff
@@ -0,0 +1,324 @@
+                const response = await certificateManagement.getOwnEnrollmentId();
+                const result = handler.handleResponse(response);
+                if (result) {
+                    enrollmentId.value = result.id;
+                }
+                busy.value--;
+            }
+
+            function filterOperations(ops: Operation[], filter: string): Operation[] {
+                if (filter != "") {
+                    //TODO more filtering
+                    let filteredOperations = ops.filter(
+                        (op) =>
+                            op.operationId.replace(/\s/g, "").toLowerCase().includes(filter) ||
+                            op.initiator.replace(/\s/g, "").toLowerCase().includes(filter) ||
+                            op.transactionInfo.parameters.toString().replace(/\s/g, "").toLowerCase().includes(filter)
+                    );
+                    return filteredOperations;
+                }
+                return ops;
+            }
+            const filteredOperations = computed(() => {
+                let filter = message.value.replace(/\s/g, "").toLowerCase();
+                if (filter != "") {
+                    //TODO more filtering
+                    let filteredOperations = operations.value.filter(
+                        (op) =>
+                            op.operationId.replace(/\s/g, "").toLowerCase().includes(filter) ||
+                            op.initiator.replace(/\s/g, "").toLowerCase().includes(filter) ||
+                            op.transactionInfo.parameters.toString().replace(/\s/g, "").toLowerCase().includes(filter)
+                    );
+                    return filteredOperations;
+                }
+                return operations.value;
+            });
+
+            const pendingOwnOperations = computed(() => {
+                let filter = message.value.replace(/\s/g, "").toLowerCase();
+                return filterOperations(operations.value, filter).filter(
+                    (op) => op.state == OperationStatus.PENDING && op.initiator == enrollmentId.value
+                );
+            });
+            const finishedOperations = computed(() => {
+                let filter = message.value.replace(/\s/g, "").toLowerCase();
+                return filterOperations(operations.value, filter).filter(
+                    (op) => op.state == OperationStatus.REJECTED || op.state == OperationStatus.FINISHED
+                );
+            });
+            const shownActionRequiredOperations = computed(() => {
+                let filter = message.value.replace(/\s/g, "").toLowerCase();
+                return filterOperations(actionRequiredOperations.value, filter).filter(
+                    (op) =>
+                        op.state == OperationStatus.PENDING &&
+                        (op.missingApprovals.users.includes(enrollmentId.value) || op.missingApprovals.groups.includes(role.value))
+                );
+            });
```

Comment:
the api provides filters, I think it would be nice to let lagom do the comparison (all operations can be extremely large for admins).

of course, to reduce waiting time for I/O, we should resolve all calls in parallel (see above)

Reply:
These filter work with the operations, that we get by fetching the watchlist (except shownActionRequired, there I already do another API call). But we could implement addtional calls with filters `getOperations(selfInitiated: false, selfActionRequired: false, states: [PENDING], watchlistOnly: true)` for pending ops and `getOperations(selfInitiated: false, selfActionRequired: false, states: [FINISHED, REJECTED], watchlistOnly: true)` for finished ops

Reply:
I implemented that in https://github.com/upb-uc4/ui-web/pull/785/commits/991a986ef6602beb685765efeb8029f0e82d6149 , please have a look.

Note: there are some weird operations on the watchlist, I guess they are resulting from testing the API?